lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -1252,6 +1256,8 @@ RequirementCheckResult TypeChecker::checkGenericArguments(
       case RequirementCheckResult::UnsatisfiedDependency:
       case RequirementCheckResult::Failure:
       case RequirementCheckResult::SubstitutionFailure:
+        if (listener)
+          listener->unsatisfiedRequirement(rawReq, firstType, secondType);
```

Comment:
I don't think we only want to call `unsatisfiedRequirement` for `UnsatisfiedDependency`.